project_1/include/utils.hpp

```diff
@@ -0,0 +1,97 @@
+#pragma once
+
+#include <cstdint>
+#include <iostream>
+#include <vector>
+
+enum event_type: std::uint8_t {
+    SEND_DATA,
+    ACKNOWLEDGEMENT,
+};
+
+// Type for packet
+struct PacketHeader {
```

Comment:
I suggest to make fields of structures public to simplify. If we want to ensure that fields cannot be changed then `const` can be used

Reply:
I think that public fields may bring us non-trivial errors in future, so it is better to keep data encapsulated.